ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/BlockCreator.java

```diff
@@ -19,10 +19,16 @@
 import org.hyperledger.besu.ethereum.core.Transaction;
 
 import java.util.List;
+import java.util.Optional;
 
 public interface BlockCreator {
   Block createBlock(final long timestamp);
 
   Block createBlock(
       final List<Transaction> transactions, final List<BlockHeader> ommers, final long timestamp);
+
+  Block createBlock(
+      final Optional<List<Transaction>> maybeTransactions,
+      final Optional<List<BlockHeader>> maybeOmmers,
+      final long timestamp);
```

Comment:
I don't think this interface method and its implementations are necessary anymore.  This might be an artifact from rayonism where this was used to differentiate behavior, but at this point I think this signature is functionally identical to the non-optional version with empty lists.

will push a suggested removal

Reply:
edit: nvm.  lets just keep the optional signatures, at least for Ommers an empty list isn't the same as an empty optional.